provider/marathon/marathon.go

```diff
@@ -537,3 +676,12 @@ func parseIndex(index string, length int) (int, error) {
 
 	return parsed, nil
 }
+
+func extractServiceName(varg ...string) string {
+	switch {
+	case len(varg) == 1:
+		return varg[0]
+	default:
```

Comment:
`extractServiceName` method is a helper to help retrieve the optional service name from the varargs.  I was trying to keep the existing template method signatures backwards compatible, and since Go does not support method overloading, the only option was to declare last argument of those methods as a vararg string.  So by convention, if there is only 1 element in the vararg, it represents a ServiceName, otherwise it's not a service name we can use, so empty string for service name is returned.